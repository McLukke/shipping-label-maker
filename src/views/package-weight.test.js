import { render, screen, act, waitFor } from "@testing-library/react";
import PackageWeight from "./package-weight";
import ShallowRenderer from "react-test-renderer/shallow";
import renderer from "react-test-renderer";
import user from "@testing-library/user-event";

describe("Package Weight", () => {
  let props;
  const getWeight = () => screen.getByRole("textbox", { name: /weight/i });
  const clickNextButton = () =>
    user.click(screen.getByRole("button", { name: /Next/i }));

  beforeEach(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });

    props = {
      updateAppState: jest.fn(),
      setCurrentStep: jest.fn(),
      currentStep: 2,
    };

    render(<PackageWeight {...props} />);
  });

  it("renders without crashing", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<PackageWeight {...props} />);
    const view = renderer.getRenderOutput();

    expect(view).toBeTruthy();
  });

  it("snapshots to match", () => {
    const component = renderer.create(<PackageWeight {...props} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("submit is called", () => {
    act(() => user.type(getWeight(), "123"));
    clickNextButton();
  });

  it("has required input", async () => {
    clickNextButton();

    await waitFor(() => {
      expect(getWeight()).toHaveClass("ant-input-status-error");
    });
  });
});
