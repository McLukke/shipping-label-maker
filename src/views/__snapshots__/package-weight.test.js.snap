// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package Weight snapshots to match 1`] = `
<div
  className="ant-row ant-row-center ant-row-middle"
  style={Object {}}
>
  <div
    className="ant-col ant-col-24"
    style={Object {}}
  >
    <h3
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Enter Package Weight (in grams):
    </h3>
  </div>
  <div
    className="ant-col ant-col-xs-24 ant-col-sm-20 ant-col-md-18 ant-col-lg-16 ant-col-xl-12"
    style={Object {}}
  >
    <form
      className="ant-form ant-form-horizontal"
      id="package-weight"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className="ant-form-item-required"
            htmlFor="package-weight_weight"
            title="Package Weight (g)"
          >
            Package Weight (g)
          </label>
        </div>
        <div
          className="ant-col ant-col-16 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="package-weight_weight"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                className="ant-space ant-space-horizontal ant-space-align-center"
                style={Object {}}
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginRight": 8,
                    }
                  }
                >
                  <button
                    className="ant-btn ant-btn-default"
                    onClick={[Function]}
                    type="button"
                  >
                    <span>
                      Back
                    </span>
                  </button>
                </div>
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    type="submit"
                  >
                    <span>
                      Next
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
